### DISCUSSION TITLE
Gang Scheduling Jobs on Airflow

### DISCUSSION THREAD

**Discussion by valayDave** [apache/airflow#36344] [Q&A]
Is there a way to get gang scheduling semantics for jobs orchestrated via Airflow? Meaning,

**Comment by potiuk** [comment]
If you are talking about https://en.wikipedia.org/wiki/Gang_scheduling then nope.  No such feature. 

And it seems it would require quite an implementation and design to allow for something like this (currently schduler is purely task and dependency based - i.e. a task can depend only on result of previous tasks and there is never a coordination between parallel running tasks). So if you would like to contribute it, I think writing an Airflow Improvement Proposal how you would like to do it.

https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals
